src/impl/graph.hpp

```diff
@@ -2,13 +2,22 @@
 #define IMPL_GRAPH_HPP
 
 #include <edge.hpp>
+#include <graph_traits.hpp>
 
 #include <map>
 #include <set>
 #include <utility>
 
 namespace graph_impl {
 
+template<typename Node>
+class Net : public graph::GraphTrait {
+public:
+    Net(Node source, Node sink) :
+        source(std::move(source)), sink(std::move(sink)) {}
+    Node source, sink;
```

Comment:
Do we really want a copy, not a reference/pointer?

Reply:
Yes. It makes more sense in my mind.
For example:
```C++
struct CustomNode {
    CustomNode(int id, int dp = 0) : id(id), dp(dp) {}
    int id;
    int dp;

    friend bool operator<(const CustomNode& lhn, const CustomNode& rhn) {
        return lhn.id < rhn.id;
    }
}
```
then
```C++
graph::Graph<CustomNode, graph::Net>g{{{1, 1}, {2, 1}, {3, 1}}, {1, 3}};
```
totally makes sense.

Reply:
Actually we kinda want reference/pointer but it should point on Node in container inside the Graph class.
I'll think about it.